### ISSUE TITLE
Regression of #763: `ansible-lint` warn if not running in a directory that is a git repo

### ISSUE THREAD

**Issue by zigarn** [ansible-community/ansible-lint#1410]
##### Summary

Regression of #763: `ansible-lint` warns if not running in a directory that is a git repo

No warning in 4.3.7, warning since 5.0.0.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
$ ansible --version
ansible 2.10.6
  config file = /home/vagrant/.ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]

$ ansible-lint --version
ansible-lint 5.0.2 using ansible 2.10.6
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Ubuntu 20.04.2 LTS (Focal Fossa) in Vagrant

##### STEPS TO REPRODUCE

1. Create a folder not in a git directory (e.g. `/tmp/test`)
2. Create a playbook in this directory
3. Launch ansible-lint against the playbook file:

```console (paste below)
$ ansible-lint playbook.yml
WARNING  Failed to discover yaml files to lint using git: fatal: not a git repository (or any of the parent directories): .git
```

##### Desired Behaviour

ansible-lint does not warn as there are files to analyze.

##### Actual Behaviour

Get a warning that as no sense about not being in a git repository.

```console (paste below)
$ ansible-lint playbook.yml
WARNING  Failed to discover yaml files to lint using git: fatal: not a git repository (or any of the parent directories): .git
```

**Comment by ansiblejunky** [comment]
Additional note here on this issue, I noticed this warning can show up multiple times. For example, I have this structure and ansible-lint generates that warning twice.
```
$ tree
.
└── my_namespace
    └── my_collection
        ├── README.md
        ├── docs
        ├── galaxy.yml
        ├── playbooks
        │   └── play.yml
        ├── plugins
        │   └── README.md
        └── roles
```

Output...
```
$ ansible-lint
WARNING  Failed to discover yaml files to lint using git: fatal: not a git repository (or any of the parent directories): .git
WARNING  Failed to discover yaml files to lint using git: fatal: not a git repository (or any of the parent directories): .git
WARNING  Listing 1 violation(s) that are fatal
my_namespace/my_collection/playbooks/play.yml:9: unnamed-task All tasks should be named
You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - unnamed-task  # All tasks should be named
Finished with 1 failure(s), 0 warning(s) on 2 files.
```

Using ansible-lint v5.0.4
